lib/authz/plugins/resources.go

```diff
@@ -0,0 +1,103 @@
+package plugins
+
+import (
+	"fmt"
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/types"
+)
+
+type Role string
+
+var (
+	RoleUser  = Role("user")
+	RoleAdmin = Role("admin")
+)
+
+type DummyUser struct {
+	Name string
+	Owns string
```

Comment:
So, the model that I had in mind is that:

a) A user is part of a set of groups. Example: carlo is part of astore-users, astore-admins, docker-pushers, system-builders, ...
b) Groups give various privileges on services. Eg, astore-user allows to read/write/list/... artifacts in astore, astore-admins can access the astore secret key to build it, or gcloud role to take job up/down, ...
c) groups (or names) can then be used by individual services. Eg, directory '/eda/' in astore is writeable by members of the group build-tool-maintainers, readable by everyone.

Given what I just typed above...
1) Shouldn't user have multiple groups? Eg, for each user, we have a set of groups he's part of.
2) Shouldn't the rego file be built around those groups?